homeassistant/components/hunterdouglas_powerview/sensor.py

```diff
@@ -55,26 +48,24 @@ class PowerviewSensorDescription(
 SENSORS: Final = [
     PowerviewSensorDescription(
         key="charge",
-        device_class=SensorDeviceClass.BATTERY,
-        native_unit_of_measurement=PERCENTAGE,
-        native_value_fn=lambda shade: round(
-            shade.raw_data[SHADE_BATTERY_LEVEL] / SHADE_BATTERY_LEVEL_MAX * 100
-        ),
-        create_sensor_fn=lambda shade: bool(
-            shade.raw_data.get(ATTR_BATTERY_KIND) != BATTERY_KIND_HARDWIRED
-            and SHADE_BATTERY_LEVEL in shade.raw_data
-        ),
+        device_class_fn=lambda shade: SensorDeviceClass.BATTERY,
+        native_unit_fn=lambda shade: PERCENTAGE,
+        native_value_fn=lambda shade: shade.get_battery_strength(),
+        create_entity_fn=lambda shade: shade.is_battery_powered(),
         update_fn=lambda shade: shade.refresh_battery(),
     ),
     PowerviewSensorDescription(
         key="signal",
         translation_key="signal_strength",
         icon="mdi:signal",
-        native_unit_of_measurement=PERCENTAGE,
-        native_value_fn=lambda shade: round(
-            shade.raw_data[ATTR_SIGNAL_STRENGTH] / ATTR_SIGNAL_STRENGTH_MAX * 100
-        ),
-        create_sensor_fn=lambda shade: bool(ATTR_SIGNAL_STRENGTH in shade.raw_data),
+        device_class_fn=lambda shade: SensorDeviceClass.SIGNAL_STRENGTH
```

Comment:
Please don't use lambdas that span more than one line. Use a separate function in that case.

Reply:
No worries - I'll update that